config/config.go

```diff
@@ -368,62 +370,23 @@ func CreateVMConfig(cfg *Config, index int) (*vm.Config, error) {
 func checkUnknownFields(data []byte) (string, error) {
 	// While https://github.com/golang/go/issues/15314 is not resolved
 	// we don't have a better way than to enumerate all known fields.
-	var fields = []string{
-		"Name",
-		"Http",
-		"Rpc",
-		"Workdir",
-		"Vmlinux",
-		"Kernel",
-		"Tag",
-		"Cmdline",
-		"Image",
-		"Cpu",
-		"Mem",
-		"Sshkey",
-		"Bin",
-		"Bin_Args",
-		"Debug",
-		"Output",
-		"Hub_Addr",
-		"Hub_Key",
-		"Dashboard_Addr",
-		"Dashboard_Key",
-		"Syzkaller",
-		"Type",
-		"Count",
-		"Devices",
-		"Procs",
-		"Cover",
-		"Reproduce",
-		"Sandbox",
-		"Leak",
-		"Enable_Syscalls",
-		"Disable_Syscalls",
-		"Suppressions",
-		"Ignores",
-		"Initrd",
-		"Machine_Type",
-		"Odroid_Host_Addr",
-		"Odroid_Slave_Addr",
-		"Odroid_Console",
-		"Odroid_Hub_Bus",
-		"Odroid_Hub_Device",
-		"Odroid_Hub_Port",
+	structType := reflect.ValueOf(Config{}).Type()
+	fields := make([]string, 0)
+	for i := 0; i < structType.NumField(); i++ {
+		field := structType.Field(i)
+		if field.Tag.Get("json") == "-" {
+			continue
+		}
+		fields = append(fields, strings.ToLower(field.Name))
 	}
+	sort.Strings(fields)
 	f := make(map[string]interface{})
 	if err := json.Unmarshal(data, &f); err != nil {
 		return "", fmt.Errorf("failed to parse config file: %v", err)
 	}
 	for k := range f {
-		ok := false
-		for _, k1 := range fields {
-			if strings.ToLower(k) == strings.ToLower(k1) {
-				ok = true
-				break
-			}
-		}
-		if !ok {
+		index := sort.SearchStrings(fields, strings.ToLower(k))
+		if index >= len(fields) || fields[index] != strings.ToLower(k) {
```

Comment:
I think it will be cleaner if you make fields `make(map[string]bool)`. No need to sort and no need to remember what exactly `SearchStrings` returns.

Also, while you are here, do you mind changing `checkUnknownFields` signature to return just single `error`? Not sure why I made it return field name when it could return `fmt.Errorf("unknown field '%v' in config", k)`.